src/Identifier/PasswordIdentifier.php

```diff
@@ -130,51 +142,21 @@ protected function _findUser($identifier, $password = null)
      * Get query object for fetching user from database.
      *
      * @param string $identifier The username/identifier.
-     * @return \Cake\ORM\Query
+     * @return \ArrayAccess|null
      */
-    protected function _query($identifier)
+    protected function _findUser($identifier)
     {
-        $config = $this->_config;
-        $table = $this->tableLocator()->get($config['userModel']);
-
-        $options = ['conditions' => $this->_buildConditions($identifier, $table)];
-
-        $finder = $config['finder'];
-        if (is_array($finder)) {
-            $options += current($finder);
-            $finder = key($finder);
-        }
-
-        if (!isset($options['username'])) {
-            $options['username'] = $identifier;
-        }
-
-        return $table->find($finder, $options);
-    }
-
-    /**
-     * Build query conditions.
-     *
-     * @param string $identifier The username/identifier.
-     * @param \Cake\ORM\Table $table Table instance.
-     * @return array
-     */
-    protected function _buildConditions($identifier, $table)
-    {
-        $usernameFields = $this->config('fields.username');
-
-        if (is_array($usernameFields)) {
-            $conditions = [];
-            foreach ($usernameFields as $field) {
-                $conditions[$table->aliasField($field)] = $identifier;
-            }
+        $fields = $this->getConfig('fields.username');
+        foreach ((array)$fields as $field) {
             $conditions = [
-                'OR' => $conditions
+                $field => $identifier
```

Comment:
@ADmad we can't support `OR` syntax for every ORM available.

Although we could change method definition of `DatasourceInterface`:
```php
public function find(string|array $field, string|array $value);
```
For our ORM that could be implemented something like that:
```php
$conditions = [
    'OR' => [
        'field_1 in' => $values
        ...
    ]
];